sympy/printing/pretty/tests/test_pretty.py

```diff
@@ -7561,3 +7561,7 @@ def test_diffgeom():
     assert pretty(rect) == "rect"
     b = BaseScalarField(rect, 0)
     assert pretty(b) == "x"
+
+def test_issue_21814():
+    assert upretty(Mul(x + y, Rational(1, 2), evaluate=False)) == 'x + y\n─────\n  2  '
+    assert upretty(Mul(Rational(1, 2), x + y, evaluate=False)) == 'x + y\n─────\n  2  '
```

Comment:
These tests could go with the others:
https://github.com/sympy/sympy/blob/4ab830d193a9d2c197d4e95ee031ca0865d011e0/sympy/printing/pretty/tests/test_pretty.py#L954-L998